katz/src/main/kotlin/katz/data/Try.kt

```diff
@@ -80,18 +80,21 @@ sealed class Try<out A> {
     /**
      * The `Failure` type represents a computation that result in an exception.
      */
-    class Failure<out A>(val exception: Throwable) : Try<A>() {
-        override val isFailure: Boolean = false
-        override val isSuccess: Boolean = true
+    data class Failure<out A>(val exception: Throwable) : Try<A>() {
+        override val isFailure: Boolean = true
+        override val isSuccess: Boolean = false
     }
 
     /**
      * The `Success` type represents a computation that return a successfully computed value.
      */
-    class Success<out A>(val value: A) : Try<A>() {
-        override val isFailure: Boolean = true
-        override val isSuccess: Boolean = false
+    data class Success<out A>(val value: A) : Try<A>() {
+        override val isFailure: Boolean = false
+        override val isSuccess: Boolean = true
     }
+
+    data class PredicateException(override val message: String) : Throwable(message)
```

Comment:
let's make this a sealed hierarchy so consumers matching over the exceptions get full compiler support to deal with all potential datatypes.